app/wss.js

```diff
@@ -17,21 +17,44 @@ const sendToAllClients = (data) => {
         data = JSON.stringify(data);
     }
     webSockServer.clients.forEach(client => {
+        console.log("Sending data to client", data);
         client.send(data);
     });
 }
+
+const validateData = (username, gameId, actionName) => {
+    // TO DO: Currently just checks username is in gameId & if they have the role to do the action, we should also check the validation-cookie
+    const playerRole = lookUpRole(gameId, username);
+    if (playerRole === null) {
+        return false;
+    }
+    return (playerRole === ROLES.HUMAN && actionName in HUMAN_ACTIONS) || (playerRole === ROLES.AI && actionName in AI_ACTIONS)
+}
+
 const onReceiveDataFromClient = (byteData) => {
     let data = JSON.parse(byteData.toString());
     let action = data.action;
+    let username = data.username;
+    let gameId = data.gameId;
     let currentRoom;
+
+    console.log("WSS Receive :", data);
+
+    const isValid = validateData(username, gameId, action.name);
+    console.log("isValid request?", isValid);
+    if (!isValid) {
+        return;
+    }
+    const game = lookUpGame(gameId);
+
     switch(action.name) {
-        case "enterRoom":
-            enterRoom(action.args.room);
-            currentRoom = getCurrentRoom();
+        case HUMAN_ACTIONS.enterRoom:
+            game.enterRoom(action.args.room);
+            currentRoom = game.getCurrentRoom();
             sendToAllClients(HumanRoomUpdateData(currentRoom));
             break;
-        case "getCurrentRoom":
-            currentRoom = getCurrentRoom();
+        case HUMAN_ACTIONS.getCurrentRoom:
+            currentRoom = game.getCurrentRoom();
```

Comment:
This is temporary, right? I think you're planning to move it back into GAME or wherever

Reply:
Yeah this gotta go, this giant switch is gonna have to go somewhere, haven't decided where, gotta finish brian stuart's assignment